libr/util/astr.c

```diff
@@ -280,7 +301,11 @@ R_API RASN1String *r_asn1_stringify_oid (const ut8* buffer, ut32 length) {
 		}
 		++i;
 	} while (X509OIDList[i].oid && X509OIDList[i].name);
-	return r_asn1_create_string (str, true, ASN1_OID_LEN);
+	RASN1String* asn1str = r_asn1_create_string (str, true, ASN1_OID_LEN);
+	if (!asn1str) {
```

Comment:
as i said, r_asn1_create_string() should take const char * as 1st argument. and the caller should always free the str. dont make ownership behaviour conditional because that confuses developers and makes it easier to introduce bugs

Reply:
`r_asn1_create_string` returns `NULL` on failure and doesn't free the string, since `str` is allocated locally in this function, not freeing it in that case would cause a leak in memory.